### PR TITLE
Kraut Space Magic™ caseless gun, and other gun changes

### PR DESCRIPTION
**Pull request by SpadesNeil**
This is probably going to need a changelog but I forget how to do that. Somebody guide me through that later.

# So first, the stuff that requires explaination

- 5.45 and 7.62 are now .223 and .308 respectively.

Why am I using caliber as opposed to millimeter? To avoid confusion. What if we _want_ to have 5.45mm in the code one day? What about the SVD that uses 7.62x54R? How confusing would it be to, at a glance, simultaneously have a gun chambered in 5.56? And furthermore, we **already** have a problem where every rifle that has 7.62mm bullets is assumed to be the same caliber, _even though there's at least 3 different kinds of 7.62mm, **none** of which are compatible!_ When I say .308, future devs and players know **exactly** what kind of bullet this gun is supposed to take. Not AK ammunition, not Mosin ammunition, but the commonplace .308 cartridge.

- SVD now uses "high velocity" 7.62x54mmR ammunition.

"High velocity" is what I am calling sniper ammo, and I am calling it that because "sniper ammo" is hitscan. This makes the SVD no longer absolutely useless as a scoped gun. This feature, which isn't new, but has so far been unique to the PTR, means that we have a chance to test it in practice and see how it works. Maybe the DMR should use hitscan ammo in the future.

- Added some comments under bullets.dm to give devs an idea of what bullet they should use for their new cartridges.

It was kind of stupid that we had 7.62 and 5.45 projectiles, when players never ever actually see the name of those projectiles anywhere at any time. I altered this so that instead a dev can use a small set of only three or four projectiles for a _wide_ variety of ammo cartridges, but they consistently know what to use for whichever gun they are making. Why would I add a new projectile for 7.62x54mmR when the one for .308 does exactly the same damage? I also added some small caliber stuff that could be used later.

- Added the "prototype caseless rifle"

Like the advanced SMG, research can build them, but they don't _spawn_ with ammo. You can however produce both stun rounds or live bullets for it in 30 round magazines. This gun is semi-automatic, and has about the power of a medium level pistol. **Note I still need to decide how the gun is balanced, so please lend some opinion on that or I am going to wing it.**

## And now for everything else.

- Add: "ratshot" projectile for future use. Basically harmless to people, but kills pesky mice from across the bar.
- Add: New Semi-automatic rifle, which is a semi-auto only version of the STS-35 automatic rifle (formerly assault rifle). Might make this civilian version illegally modifiable in future.
- Add: The PTR-7 now has an incendiary shell. Admin-spawnable only at this time.
- Change: 10mm changed to 10mm Auto, again to avoid future confusion if other 10mm calibers come along.
- Change: Advanced SMG are now compatible with standard 9mm magazines.
- Change: Advanced SMG is now slightly more accurate than other SMGs in the game.
- Change: Advanced SMG no longer spawns spawning with ammo preloaded. You have to make the ammo separate.
- Change: Advanced SMG's magazine are actually available in the protolathe, and justify its use of silver with the 21 round capacity.
- Change: Assault rifle renamed to automatic rifle.
- Change: Consistent naming on .45 and .223 armor-piercing magazines.
- Change: Description of Advanced SMG no longer makes reference to using armor-piercing ammo, because **it never did.**
- Change: M1 Garand now does 60 damage, but now uses .30-06, which you can't make in an autolathe. Pick your shots wisely.
- Change: Made the STS-35 lore more interesting.
- Change: R&D submachine gun is now called the "advanced SMG".
- Change: R&D's "saber" submachine gun has had several edits to it.
- Change: Reduced Advanced SMG's magazine by one bullet, formerly 22, because burst fire consistency.
- Change: The back-end naming of the Advanced SMG and relevant magazines are now consistent with how other things are named, and can't get confused.
- Change: The traitor menu for weapons will also show the caliber of the gun you are about to buy.
- Change: Tommygun corrected to Tommy Gun.
- Fix: Advanced SMG's magazine sprite are no longer broken.
- Fix: The Vintage Repeater's icons are no longer broken.
- Removed: Obsolete 9mm ammo box deleted from the game.

ToDo:
- [ ] Changelog. How the fuck do I make changelogs again? :U

### CODE REVIEW

**Comment by Naidh** [comment]
Personally okay with the G11's existence given consideration to the fact that it's actually something interesting and it's provided as a show-of-features for something else that doesn't make for suddenly getting forty-five hundred different varieties of Uncle Jimbob's fucked-up AR-15.

**Comment by SpadesNeil** [comment]
That was why I chose the G11 as the inspiration. In-lore, it's not a G11. The G11 just happens to be the appearance I stole, just as the STS-35 is an AK rip off.

If somebody else wants to sprite up a G11 look-alike that is more visually unique, be my guest, but this one already has all the bells and whistles of hand sprites and everything else.

I also picked the G11 because what other 21st century gun would be appropriate than the original Kraut Space Magic™ rifle?

File: maps/southern_cross/backup/southern_cross-6.dmm [hunk 1]
```diff
@@ -0,0 +1,2502 @@
+"aa" = (/turf/space,/area/space)
```
Comment (SpadesNeil) [inline maps/southern_cross/backup/southern_cross-6.dmm]: Oops. This wasn't supposed to happen.

File: maps/southern_cross/southern_cross-6.dmm [hunk 2]
```diff
@@ -215,13 +215,13 @@
 "eg" = (/obj/structure/table/rack,/obj/item/ammo_casing/rocket,/obj/item/ammo_casing/rocket,/obj/item/ammo_casing/rocket,/obj/item/ammo_casing/rocket,/obj/item/ammo_casing/rocket,/obj/item/ammo_casing/rocket,/turf/unsimulated/floor{icon_state = "dark"},/area/centcom/specops)
 "eh" = (/obj/machinery/door/blast/regular{icon_state = "pdoor1"; id = "ASSAULT"; name = "Assault Weapon Storage"; p_open = 0},/turf/unsimulated/floor{icon_state = "dark"},/area/centcom/specops)
 "ei" = (/turf/unsimulated/floor{icon_state = "dark"},/area/centcom/specops)
-"ej" = (/obj/structure/table/rack,/obj/item/ammo_magazine/m762,/obj/item/ammo_magazine/m762,/obj/item/ammo_magazine/m762,/obj/item/ammo_magazine/m762,/obj/item/ammo_magazine/m762,/obj/item/ammo_magazine/m762,/turf/unsimulated/floor{icon_state = "dark"},/area/centcom/specops)
+"ej" = (/obj/structure/table/rack,/obj/item/ammo_magazine/m308,/obj/item/ammo_magazine/m308,/obj/item/ammo_magazine/m308,/obj/item/ammo_magazine/m308,/obj/item/ammo_magazine/m308,/obj/item/ammo_magazine/m308,/turf/unsimulated/floor{icon_state = "dark"},/area/centcom/specops)
```
Comment (SpadesNeil) [inline maps/southern_cross/southern_cross-6.dmm]: That's much better.

**Comment by Cerebulon** [comment]
For your first point, personally I'm not sure we need hundreds of unique ammo types for TechnicalAccuracy, there's already a lot and please god some standardization in the future.
I don't think it's a "problem" that three bullets that are technically different in real life are all compatible in the game, because it's a game and it should be accessible to people who aren't gun nerds.

**Comment by SpadesNeil** [comment]
The fuck is going on here?
image

**Comment by SpadesNeil** [comment]
Aside from that Travis error, this will work now. Fully tested.

**Comment by parasitoide** [comment]
I quite prefer the millimeter naming scheme, and would prefer if we found some other way to differentiate- .308 means less to me than 7.62 does, not knowing guns as well as I should.

**Comment by SpadesNeil** [comment]
>In my honest opinion, naming projectiles according to the sort of cartridge they come from makes far more sense than using generic cartridges that are expected to match up to potentially multiple different types of guns. If we need to balance one particular gun's damage, then making it share a generic cartridge with 15 other guns makes that exceptionally difficult.

Well, this is how other guns in the game are currently done, and until that changes, we may as well be consistent about it. We can always alter it later, but if we do, we have to refactor **all** of the guns including pistols which have already been using this system. I just brought the rifle calibers in line with this way of doing it.

>Also, if you introduce a gun with "knockoff of [real-life gun]", even if you explicitly state multiple times in the pr that it really isn't that gun, we're gonna think it's that gun. It would be far easier for you to exclude that mention if it really isn't the gun you're comparing it to, so that we don't make that connection.

Fine, but I'm keeping this photo down here. If I have to change the icon names before merging that's fine too.

image

**Comment by SpadesNeil** [comment]
>I quite prefer the millimeter naming scheme, and would prefer if we found some other way to differentiate- .308 means less to me than 7.62 does, not knowing guns as well as I should.

I felt that way previously, but realized that method is worthless when so many guns use the same millimeter.

For example, if we convert .45 to that, now it's 11.43×23mm, and .357 Magnum is now 9×33mmR.

I'm going by what the ammo types are most commonly known as. 5.56mm would have also been acceptable for .223 but 7.62 was not acceptable for .308 as 7.62 refers to at least three rifle calibers and one pistol caliber.

**Comment by parasitoide** [comment]
That's actually a good point IRT 7.62 because I'm familiar with both 7.62 Russian and 7.62 NATO, but for 5.56 I'm fairly sure its unique, and more importantly when it comes to item names, I don't ever hear it referred to as _.223_.

**Comment by SpadesNeil** [comment]
Well, let's hear second opinions.

As I said, making it 5.56 again is totally fine, I only did it because I foresee confusion in the future.

For 7.62/.308 however, like I said, and I'm glad you see my point, it's too confusing for _that_ one.

**Comment by SpadesNeil** [comment]
>please no more 20th century firearms, we have 500 years of gun culture almost totally unaccounted for

Inspirin' me to toss in fancy looking guns that are totally fictional. :B

**Comment by elgeonmb** [comment]
my California Girl Fears Guns eyes are glazing over all these numbered gauges and dimensions and shit; i come from a simpler time when there were like 5 ammo pickups named after five wildly different kinds of firearm. like i'm not saying you have to cater to me but I am the lowest common denominator here

**Comment by SpadesNeil** [comment]
>my California Girl Fears Guns eyes are glazing over all these numbered gauges and dimensions and shit; i come from a simpler time when there were like 5 ammo pickups named after five wildly different kinds of firearm. like i'm not saying you have to cater to me but I am the lowest common denominator here

be gone fudd

**Comment by SpadesNeil** [comment]
And yes, staff input required.

- I suspect the M1 Garand being buffed in ammo and nerfed in the ability to _get_ ammo is going to be a topic of contention.
- I haven't actually added the prototype rifle to the protolathe yet because I don't know how much metal it should cost. Obviously the ammo will require a generous amount of phoron though.
- How much ammo is too much ammo for the prototype rifle?
- Are the changes I made to the SMG okay?

Just some of the stuff that needs discussion.

**Comment by Anewbe** [comment]
As discussed at the Staff Meeting, and in accordance with our rules, this PR needs to be split into multiple others. One feature per PR is requested, so we can discuss them on their individual merits.

**Pull request closed without merging**